internal/store/annotation.go

```diff
@@ -0,0 +1,247 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+package store
+
+import (
+	"database/sql"
+	sq "github.com/Masterminds/squirrel"
+	"github.com/mattermost/mattermost-cloud/model"
+	"github.com/pkg/errors"
+)
+
+var annotationSelect sq.SelectBuilder
+var annotationColumns = []string{
+	"Annotation.ID", "Annotation.Name",
+}
+
+func init() {
+	annotationSelect = sq.Select(annotationColumns...).
+		From("Annotation")
+}
+
+// GetAnnotationByName fetches the given annotation by name.
+func (sqlStore *SQLStore) GetAnnotationByName(name string) (*model.Annotation, error) {
+	return sqlStore.getAnnotationByName(sqlStore.db, name)
+}
+
+func (sqlStore *SQLStore) getAnnotationByName(db queryer, name string) (*model.Annotation, error) {
+	var annotation model.Annotation
+
+	builder := annotationSelect.
+		Where("Name = ?", name).
+		Limit(1)
+	err := sqlStore.getBuilder(db, &annotation, builder)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, nil
+		}
+		return nil, errors.Wrap(err, "failed to get annotation by name")
+	}
+
+	return &annotation, nil
+}
+
+// CreateAnnotation records the given annotation to the database, assigning it a unique ID.
+func (sqlStore *SQLStore) CreateAnnotation(annotation *model.Annotation) error {
+	return sqlStore.createAnnotation(sqlStore.db, annotation)
+}
+
+func (sqlStore *SQLStore) createAnnotation(db execer, annotation *model.Annotation) error {
+	annotation.ID = model.NewID()
+
+	_, err := sqlStore.execBuilder(db, sq.Insert("Annotation").
+		SetMap(map[string]interface{}{
+			"ID":   annotation.ID,
+			"Name": annotation.Name,
+		}))
+	if err != nil {
+		return errors.Wrap(err, "failed to create annotation")
+	}
+
+	return nil
+}
+
+// getOrCreateAnnotations fetches annotations by name or creates them if they do not exist.
+func (sqlStore *SQLStore) getOrCreateAnnotations(db dbInterface, annotations []*model.Annotation) ([]*model.Annotation, error) {
+	for i, ann := range annotations {
+		annotation, err := sqlStore.getOrCreateAnnotation(db, ann)
+		if err != nil {
+			return nil, err
+		}
+		annotations[i] = annotation
+	}
+
+	model.SortAnnotations(annotations)
```

Comment:
It is not really important, it is done mainly for convenience and testing.

Reply:
If it doesn't matter, I think we should skip sorting them, as there's always a runtime performance hit for sorting

Reply:
Well, I do not think that it would impact performance in a significant way as it is hard for me to imagine there being more than 10 annotations on a single installation/cluster, but I agree if that is not really needed it can be removed.